### DISCUSSION TITLE
New components

### DISCUSSION THREAD

**Discussion by juan-manuel-alberro** [mantinedev/mantine#10] [Feedback]
Hi guys, first of all, good job with this UI library, looks super promising!

My question is, there's any plan for the near/medium term to include other components like

- Accordion
- Sliders
- Grid system (columns/rows)
- Date picker/time picker
- Wizards/Steppers
- Timelines

Thank you!

**Comment by rtivital** [comment]
Hi Juan, thanks for reaching out!

I have plans on adding components and hooks incrementally over time, to track progress I've made [Mantine trello](https://trello.com/b/R0D766tA/mantine) public. Currently there are tasks that I want to publish in next minor and major releases. 

Regarding your list components which are not on the board:
- Wizards/Steppers/Timelines – I've never found any of these that will fit my needs, each time it is a custom solution for each project, so I think there is no point in adding them. 
- Date/time picker will be added in next major release (2.0.0) in separate package (@mantine/dates) as they will depend on date-fns or dayjs library and not all projects require date inputs. 

About time picker – please share your idea on how to implement this, I've used the one on the picture, but it feels very bloated:
<img width="98" alt="Screen Shot 2021-05-04 at 3 32 27 PM" src="https://user-images.githubusercontent.com/10353856/117003825-0326cb00-acee-11eb-9724-fe7e491bce60.png">

**Reply by juan-manuel-alberro** [reply]
Thanks for sharing the roadmap! I agree about the wizards/timeline/etc, just asking since other UI libraries have it as well but I never use those tbh.

For date/time pickers, I like this approach and is kinda similar to the look&fell of the library 😄 

<img src="https://i.pinimg.com/originals/0b/9a/89/0b9a89760e635f8256f00088d78dece8.png" width="400"/>

**Reply by rtivital** [reply]
Hmm, looks too complicated, I'll try things out and see how it would feel, now I'm thinking that regular input with mask hh:mm:ss(:am/pm) will do the job

**Reply by cstrat** [reply]
Excited to see date/time is on the roadmap.

Agree with the comments around wizards and steppers.

With date/time - simple is great but having a popover calendar view is great too.
Maybe start with the simple input mask and see if any good pull requests come in for something more complicated.

**Reply by evangow** [reply]
@rtivital depending on your target release for v2, if you're adding a date/time picker, you might want to look at the Temporal, which is a Stage 3 proposal to do the things that date-fns, dayjs etc. do https://github.com/tc39/proposal-temporal

Not sure what stage it will be in by the time you start evaluating the various options, but this would definitely be a good one to add since it's browser-native

**Reply by rtivital** [reply]
@evangow My biggest concern about native Date features is localization, date formatting in particular. I've tried doing things with native dates and localization keeps producing random bugs, e.g. yesterday [I had to delete tests](https://github.com/mantinedev/mantine/commit/fc4d1867c17c006c70989ed1dd2bfad645fd3f08) for [this function](https://github.com/mantinedev/mantine/blob/master/src/mantine-dates/src/utils/get-weekdays-names/get-weekdays-names.ts) as they were crushing when jest was run on all tests only (no crushes on single file run). I'll go mad if it will pop out randomly like this, so my best bet is to use dayjs and forget about this random stuff.

**Comment by lyquocnam** [comment]
Same question with `Navbar` component and support responsive.

**Reply by rtivital** [reply]
Can you describe what responsive support do you need?

**Reply by lyquocnam** [reply]
Ya, i need the same `Header` component like mantine.dev document. it so beautiful !
And 1 more, i think we should have some prebuit layout. This will very useful. Thank you so much! [example]
(https://vuetifyjs.com/en/getting-started/wireframes/#examples) 

===

<img width="1438" alt="Screen Shot 2021-05-07 at 17 03 33" src="https://user-images.githubusercontent.com/8347826/117433918-3c617400-af56-11eb-945f-19a3a7aa2e02.png">
<img width="296" alt="Screen Shot 2021-05-07 at 17 03 52" src="https://user-images.githubusercontent.com/8347826/117433925-3ec3ce00-af56-11eb-9dd7-c9234402a253.png">

**Reply by rtivital** [reply]
I'll consider adding these components after 2.0.0 release, currently I do not have good understanding on how to make these components general purposed, as mantine.dev layout is very hacky with fixed positioning and paddings.

Thanks for feedback!
